### PR TITLE
Added support for Apple Silicon

### LINKED ISSUE #270: bazelisk.py does not support Apple arm64
This check here:
https://github.com/bazelbuild/bazelisk/blob/280b07a854a70f2ab0887d644e364c8e8c1c6187/bazelisk.py#L210

prevents it from working on Apple arm64.

The simplest fix is to just delete that check.  If the architecture is not supported, the file will fail to download.

### PR DESCRIPTION
**Pull request by linzhp**
Bazelish should check the Bazel version. If it is greater or equal to than 4.1, add support for arm64 so we can run Bazel on Apple Sillicon. Fixed #270

### CODE REVIEW

**Comment by mindaugasrukas** [comment]
LGTM. Works on my M1 machine.

**Comment by linzhp** [comment]
ping @philwo

**Review by philwo — APPROVED** [review]
Thank you!

File: bazelisk.py [hunk 1]
```diff
@@ -206,11 +206,22 @@ def determine_executable_filename_suffix():
 
 
 def determine_bazel_filename(version):
+    supported_machines = ["x86_64"]
+    versions = version.split('.')[:2]
+    if len(versions) == 2:
+        # released version
+        major, minor = int(versions[0]), int(versions[1])
+        if major > 4 or major == 4 and minor >= 1:
+            # Apple Sillicon was supported since 4.1:
```
Comment (philwo) [inline bazelisk.py:215]: nit: s/Sillicon/Silicon/

**Review by philwo — DISMISSED** [review]
Sorry, found two little things to change - could you have a look?

**Comment by linzhp** [comment]
@fweikert I believe the current CI errors are transient issues. It can be fixed by restarting the job, but I don't have the permission to do so.

File: bazelisk.py [hunk 2]
```diff
@@ -206,16 +206,36 @@ def determine_executable_filename_suffix():
...
+            or operating_system == "linux"
+            and (major > 3 or major == 3 and minor >= 4)
+        ):
+            # Linux arm64 was supported since 3.4.0.
+            # Darwin arm64 was supported since 4.1.0.
+            supported_machines.append("arm64")
+    elif operating_system in ("darwin", "linux"):
+        # This is needed to run bazelisk_test.sh on Linux and Darwin arm64 machines, which are
+        # becoming more and more popular.
+        # It works becasue all recent commits of Bazel supports arm64 on Darwin and Linux.
```
Comment (fweikert) [inline bazelisk.py:227]: typo

Comment (fweikert) [inline bazelisk.py:227]: support

File: bazelisk.py [hunk 3]
```diff
@@ -206,16 +206,36 @@ def determine_executable_filename_suffix():
 
 
 def determine_bazel_filename(version):
+    operating_system = get_operating_system()
```
Comment (fweikert) [inline bazelisk.py:209]: please move this block into its own function

**Review by fweikert — CHANGES_REQUESTED** [review]

**Comment by linzhp** [comment]
@fweikert addressed your comments. All tests passed.

**Pull request merged**